flink-ml-python/pyflink/ml/lib/feature/vectorassembler.py

```diff
@@ -27,21 +29,61 @@ class _VectorAssemblerParams(
     HasOutputCol,
     HasHandleInvalid
 ):
+
+    """
+    Checks the inputSizes parameter.
+    """
+    def sizes_validator(self) -> ParamValidator[Tuple[int]]:
+        class SizesValidator(ParamValidator[Tuple[int]]):
+            def validate(self, indices: Tuple[int]) -> bool:
+                if indices is None:
+                    return False
+                for val in indices:
+                    if val < 0:
+                        return False
+                return len(indices) != 0
+        return SizesValidator()
+
     """
     Params for :class:`VectorAssembler`.
     """
 
+    INPUT_SIZES: Param[Tuple[int, ...]] = IntArrayParam(
+        "input_sizes",
+        "Sizes of the input elements to be assembled.",
+        None,
+        sizes_validator(None))
+
     def __init__(self, java_params):
         super(_VectorAssemblerParams, self).__init__(java_params)
 
+    def set_input_sizes(self, *sizes: int):
+        return self.set(self.INPUT_SIZES, sizes)
+
+    def get_input_sizes(self) -> Tuple[int, ...]:
+        return self.get(self.INPUT_SIZES)
+
+    @property
+    def input_sizes(self) -> Tuple[int, ...]:
+        return self.get_input_sizes()
+
 
 class VectorAssembler(JavaFeatureTransformer, _VectorAssemblerParams):
     """
-    A Transformer which combines a given list of input columns into a vector column. Types of
-    input columns must be either vector or numerical value.
-
-    The `keep` option of :class:HasHandleInvalid means that we output bad rows with output column
-    set to null.
+     A Transformer which combines a given list of input columns into a vector column. Input columns
+     would be numerics or vectors whose size is specified by the {@link #INPUT_SIZES} parameter.
+     Invalid input data with null values or values with wrong sizes would be dealt with according to
+     the strategy specified by the {@link HasHandleInvalid} parameter as follows:
```

Comment:
`{@link ...}` is JavaDoc grammar and unsuitable for python documents. Let's change it to the way it has originally used, like `:class:HasHandleInvalid`.